lib/services/sshAdapter.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+
+var Bluebird = require("bluebird");
+
+function SshAdapter (ec2) {
+	var SSH_POLLING_INTERVAL = 30; /*Time in Seconds*/
+
+	this.checkIfSshable = Bluebird.method(function (instanceID) {
+		return this.canSsh(instanceID)
+		.then(function (sshable) {
+			if (sshable) {
+				return Bluebird.resolve();
+			}
+			else {
+				return Bluebird.delay(SSH_POLLING_INTERVAL * 1000)
+				.then(this.checkIfSshable.bind(this, instanceID));
+			}
+		}.bind(this));
+	});
+
+	this.startSshPolling = Bluebird.method(function (instanceID) {
+
+		return this.checkIfSshable(instanceID)
+		.catch(function (err) {
+			return Bluebird.reject(err);
```

Comment:
The `Bluebird.reject()` call here is unnecessary since this code is internal to a promise chain. You can simply throw the error. Furthermore, this block has no effect since it simply catches and re-throws an error without changing anything. We should probably just remove this frame.